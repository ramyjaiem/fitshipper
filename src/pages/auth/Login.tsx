import React, { useContext, useState } from "react";
import Button from "../../components/Button";
import Input from "../../components/Input";
import AuthContainer from "../../containers/AuthContainer";
import AuthLayout from "../../containers/layouts/AuthLayout";
import { AuthContext } from "../../contexts/auth.context";

interface Props {}

const Login = (props: Props) => {
  const LoginForm = () => {
    const { login, errorMessage } = useContext(AuthContext);
    const [email, setEmail] = useState<string | null>(null);
    const [password, setPassword] = useState<string | null>(null);
    return (
      <>
        <Input
          placeholder="Email"
          label="Email"
          name="email"
          type="email"
          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
            setEmail(e.target.value)
          }
        />
        <Input
          placeholder="Password"
          label="Password"
          name="password"
          type="password"
          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
            setPassword(e.target.value)
          }
        />
        <p className=" text-red-500 font-bold text-sm mt-3">{errorMessage}</p>
        <Button
          label="Login"
          className="mt-5 float-right h-auto w-full"
          onClick={() => login(email, password)}
        />
      </>
    );
  };
  return (
    <AuthLayout>
      <AuthContainer title="Login" form={<LoginForm />} />
    </AuthLayout>
  );
};

export default Login;
