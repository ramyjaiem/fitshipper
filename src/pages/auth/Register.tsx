import React from "react";
import RegisterForm from "../../components/forms/RegisterForm";
import AuthContainer from "../../containers/AuthContainer";
import AuthLayout from "../../containers/layouts/AuthLayout";

interface Props {}

const Register = (props: Props) => {
  return (
    <AuthLayout>
      <AuthContainer title="Create Account" form={<RegisterForm />} />
    </AuthLayout>
  );
};

export default Register;
