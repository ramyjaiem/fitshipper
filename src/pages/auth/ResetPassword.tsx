import ResetPasswordForm from "../../components/forms/ResetPasswordForm";
import AuthContainer from "../../containers/AuthContainer";
import AuthLayout from "../../containers/layouts/AuthLayout";

const ResetPassword = () => {
  return (
    <AuthLayout>
      <AuthContainer title="Reset Password" form={<ResetPasswordForm />} />
    </AuthLayout>
  );
};

export default ResetPassword;
