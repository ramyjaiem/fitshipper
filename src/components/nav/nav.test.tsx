import { shallow } from "enzyme";
import React from "react";
import NavBar from ".";
import { AuthContext, AuthContextMock } from "../../contexts/auth.context";

describe("Testing navbar...", () => {
  it("should render correctly with no props", () => {
    const component = shallow(<NavBar />);

    expect(component).toMatchSnapshot();
  });
  it("should dispaly name when user connected", () => {
    const TestComponent = () => (
      <AuthContext.Provider
        value={
          {
            user: { email: "Ramy Jaiem" },
          } as jest.Mocked<any>
        }
      >
        <NavBar />
      </AuthContext.Provider>
    );
    const element = shallow(<TestComponent />);
    expect(element.html()).toContain("Ramy Jaiem");
    expect(element).toMatchSnapshot();
  });
});
