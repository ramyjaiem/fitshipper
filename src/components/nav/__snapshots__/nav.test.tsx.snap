// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing navbar... should dispaly email when user connected 1`] = `ShallowWrapper {}`;

exports[`Testing navbar... should render correctly with no props 1`] = `ShallowWrapper {}`;
